### DISCUSSION TITLE
Cis/trans isomer labeling for cyclic systems

### DISCUSSION THREAD

**Discussion by luchini18** [rdkit/rdkit#6085] [Q&A]
Hi,
I know RDKit has cis/trans bond labels across double bonds but was wondering if there was simple way to describe or label cyclic alkanes as cis or trans. 

An Example:
![Screenshot 2023-02-13 at 10-33-39 13 2 Cis-Trans Isomers (Geometric Isomers) The Basics of General Organic and Biological Chemistry](https://user-images.githubusercontent.com/25237723/218531149-4d41de19-fae1-4976-8397-63021cbd9bc3.png)

R and S atomic labeling is useful for simple systems like this example, but doesn't necessarilty tell you if substituents off the ring are on the same/opposite side for more complex systems. 

Any ideas?

**Comment by GLPG-GT** [comment]
I do not have the answer in terms of code; but for conceptual clarity, could you please show examples of the 'more complex systems' you were thinking about?
In the above case, the 'cis' molecule is achiral, and its two stereoatoms are related by the reflection operation, so they cannot have the same configuration (and indeed, it's a 1R, 2S _meso_ stereoisomer); the 'trans' molecule is chiral, and in particular the one you reported is the 1R, 2R enantiomer.
Then you have cases like a 1,4-disubstituted cyclohexane, for instance, where the substituents can be on the same or opposite side of the ring, and in both cases the molecule is achiral, but the stereocenters are not related by the reflection operation. It would be interesting to see what happens to the CIP configuration.
This is what I find with an example:
![image](https://user-images.githubusercontent.com/96192927/218746658-11ab4a86-482b-4430-9825-eaa55fdc83d4.png)
![image](https://user-images.githubusercontent.com/96192927/218746789-d10376e3-0a33-41e7-84e9-51c3ebc08e70.png)
Hmm...
Also perhaps interesting, if I do not run the `Assign` code, the `Find` code returns an empty list. Even with `includeCIP = True`.

**Reply by luchini18** [reply]
Yes, okay. So with your first example if I add some heteroatoms I can manipulate the CIP labels to have the same R,R configuration as when the substituents are on the opposite side of the ring plane even if they are on the same side. 

![image](https://user-images.githubusercontent.com/25237723/218813760-cd5f0391-0640-4f11-95dd-399a65c424d2.png)4

I guess my ultimate goal is to have an automated way to designate if substituents are on the same/opposite side of the ring but I might have to get a bit more creative.

**Reply by luchini18** [reply]
Also interesting about your Assign and Find functions. My Find seems to work without the Assign call, even without the includeCIP arg. I'm using version 2022.09.1
 
![image](https://user-images.githubusercontent.com/25237723/218816543-96474ccb-95a7-45bb-a55a-7704b3bdaa9b.png)

**Reply by GLPG-GT** [reply]
Sure. My example referred specifically to the case you showed originally, which was a rather symmetric molecule with some chiral and some achiral stereoisomers, and the substituents that you wanted to identify as 'cis' or 'trans' were the same (Me).
Your new example is an asymmetric molecule, it cannot be achiral regardless of the configuration of the stereocenters.
Maybe (not tried) that is also why `FindMolChiralCenters()` works without running `Assign` first. My example was an achiral molecule, whose stereocenters were both on the plane of symmetry, hence 'r' or 's' rather than 'R' or 'S'.
For a more general case, indeed, it is probably true that the sp3 CIP configuration does not tell you anything particularly direct about the spatial relationship of the substituents you are thinking of. And I agree, David's approach is probably the most promising.

**Reply by luchini18** [reply]
Ah maybe I should have used a more specific example regarding my problem in my posting, my bad! Thanks for the response!

**Comment by DavidACosgrove** [comment]
Given that you are investigating an inherently 3D property, you could use
the embedding code to generate a 3D conformer and analyse the bond vectors
in 3D. I imagine that if the dot product of the 2 vectors is positive
they’re on the same side, if negative they’ll be on opposite sides.

On Tue, 14 Feb 2023 at 17:47, Guilian Luchini ***@***.***>
wrote:

> Also interesting about your Assign and Find functions. My Find seems to
> work without the Assign call, even without the includeCIP arg. I'm using
> version 2022.09.1
>
> [image: image]
> <https://user-images.githubusercontent.com/25237723/218816543-96474ccb-95a7-45bb-a55a-7704b3bdaa9b.png>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/rdkit/rdkit/discussions/6085#discussioncomment-4973727>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACGF2FUZZTKVQCTXR6X5INDWXPAJ5ANCNFSM6AAAAAAU2SJRDA>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>
-- 
David Cosgrove
Freelance computational chemistry and chemoinformatics developer
http://cozchemix.co.uk

**Reply by luchini18** [reply]
Hi @DavidACosgrove ,
Thanks for the idea, I think you might be on to something, I tried this out with a few molecules and sometimes it works out that most times they are "cis" compounds the dot product is positive, but sometimes it ends up being negative depending on the conformer. Perhaps some optimization would be useful here. 

This is just for 10 different conformers of the same molecule, sometimes results are positive, sometimes negative, depending on the run. In this case I normalized the vectors but that doesn't effect the sign of the resulting dot product.

![image](https://user-images.githubusercontent.com/25237723/218853467-18a91a7c-28e6-4255-8e69-909e81e0acb9.png)

**Comment by DavidACosgrove** [comment]
If I had thought it through, I would have remembered that of course the
cyclohexane ring isn’t flat as drawn in 2D. Sorry. I did learn organic
chemistry once, honest.


On Tue, 14 Feb 2023 at 20:34, Guilian Luchini ***@***.***>
wrote:

> Hi @DavidACosgrove <https://github.com/DavidACosgrove> ,
> Thanks for the idea, I think you might be on to something, I tried this
> out with a few molecules and sometimes it works out that most times they
> are "cis" compounds the dot product is positive, but sometimes it ends up
> being negative depending on the conformer. Perhaps some optimization would
> be useful here.
>
> This is just for 10 different conformers of the same molecule, sometimes
> results are positive, sometimes negative, depending on the run.
>
> [image: image]
> <https://user-images.githubusercontent.com/25237723/218853467-18a91a7c-28e6-4255-8e69-909e81e0acb9.png>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/rdkit/rdkit/discussions/6085#discussioncomment-4975384>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACGF2FS3CNRYX4IIDZVNMJ3WXPT6DANCNFSM6AAAAAAU2SJRDA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>
-- 
David Cosgrove
Freelance computational chemistry and chemoinformatics developer
http://cozchemix.co.uk

**Reply by luchini18** [reply]
Still an interesting idea. Wondering if I could use the ring atoms to define some sort of averaged plane to compare the substituent positions, not sure if all conformations would behave especially as ring size gets larger.

**Reply by GLPG-GT** [reply]
Just an idea following up from your discussion.
Can we say that in 'any' conformer the two atoms attached to the ring, so the O in OH and N in NH2 in your example, are closer in space when they are 'cis' as opposed to when they are 'trans'? For 1,4-disubstituted cyclohexanes I would think so. IIRC in 1H NMR you can sometimes tell relative stereochemistry from certain values of the coupling constants, influenced indeed by the spatial relationship between the atoms. But to be checked.
Anyway, if that is so, then perhaps you could measure the distance between the two atoms in your original molecule, then invert one of the two stereocenters (can that be done in rdkit BTW?), measure the distance again, and if it's smaller, the original molecule was trans, if it is larger, it was cis.
Could that work?

**Reply by luchini18** [reply]
Hmm yes distances could work. I'm working with conformational ensembles of cis and trans species so I guess I would maybe have a bimodal distribution of distances with the shorter cluster for the cis mols and the larger cluster for trans. 
Dihedrals may also work using the substituent atoms and the chiral centers but I'd have to see what these would look like in 3D
